client/CPlayerInterface.cpp

```diff
@@ -1642,7 +1662,8 @@ void CPlayerInterface::update()
 	}
 
 	//in some conditions we may receive calls before selection is initialized - we must ignore them
-	if (adventureInt && !adventureInt->selection && GH.topInt() == adventureInt)
+	if (adventureInt && GH.topInt() == adventureInt
+		&& (!adventureInt->selection && LOCPLINT->playerID != PlayerColor::SPECTATOR))

```

Comment:
CPlayerInterface::isSpectator is bad solution sadly since I don't want even more complex check everywhere and LOCPLINT might be not present when run headless. For now I just decide to check settings "session" everywhere since it's globally visible, but can't really say I'd like it.